jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/main/java/org/jbpm/workbench/es/client/editors/quicknewjob/NewJobViewImpl.java

```diff
@@ -0,0 +1,463 @@
+                                         getNumericInputListener(),
+                                         false);
+
+        myParametersGrid.setHeight("200px");
+        myParametersGrid.setEmptyTableWidget(new Label(constants.No_Parameters_added_yet()));
+
+        initGridColumns();
+        initDateTimePicker();
+
+        advancedContent.add(myParametersGrid);
+        inlineNotification.setType(InlineNotification.InlineNotificationType.DANGER);
+    }
+
+    protected EventListener<KeyboardEvent> getNumericInputListener() {
+        return (KeyboardEvent e) -> {
+            int keyCode = e.getKeyCode();
+            if (keyCode <= 0) { //getKeyCode() returns 0 for numbers on Firefox 53
+                keyCode = e.getWhich();
+            }
+            if (!((keyCode >= KeyCodes.KEY_NUM_ZERO && keyCode <= KeyCodes.KEY_NUM_NINE) ||
+                    (keyCode >= KeyCodes.KEY_ZERO && keyCode <= KeyCodes.KEY_NINE) ||
+                    (keyCode == KeyCodes.KEY_BACKSPACE || keyCode == KeyCodes.KEY_LEFT || keyCode == KeyCodes.KEY_RIGHT))) {
+                e.preventDefault();
+            }
+        };
+    }
+
+    public void show() {
+        cleanForm();
+        modal.show();
+    }
+
+    protected void initDateTimePicker() {
+        this.dateRangePicker = dateRangePickerProvider.get();
+        dateRangePicker.getElement().setReadOnly(true);
+        dateRangePicker.getElement().setAttribute("placeholder",
+                                                  Constants.INSTANCE.Due_On());
+        dateRangePicker.getElement().getClassList().add("form-control");
+
+        final Div div = (Div) getDocument().createElement("div");
+        div.setAttribute("data-filter",
+                         "Date");
+        div.getClassList().add("input-group");
+        div.getClassList().add("filter-control");
+        div.appendChild(dateRangePicker.getElement());
+        dateFiltersInput.appendChild(div);
+        dateFiltersInput.setHidden(true);
+    }
+
+    protected void resetDateTimePicker() {
+        final DateRangePickerOptions options = getDateRangePickerOptions();
+        dateRangePicker.setup(options,
+                              null);
+        dateRangePicker.addApplyListener((e, p) -> {
+            selectedDate = p.getStartDate().milliseconds(0).asDate();
+            dateRangePicker.getElement().setAttribute("placeholder",
+                                                      DateUtils.getDateTimeStr(selectedDate));
+        });
+    }
+
+    protected DateRangePickerOptions getDateRangePickerOptions() {
+        final DateRangePickerOptions options = DateRangePickerOptions.create();
+        options.setAutoUpdateInput(false);
+        options.setAutoApply(true);
+        options.setTimePicker(true);
+        options.setTimePickerIncrement(1);
+        options.setSingleDatePicker(true);
+        options.setMinDate(moment());
+
+        return options;
+    }
+
+    boolean validateForm() {
```

Comment:
Can we move this validation to Presenter?
This type of validation was a common generator of negative Selenium tests, which we want to avoid and test via unit tests.

One of the options is to move it under createJob(...) method in presenter, as a whole or each validation divided into it's own method, and accompany it with a couple of more methods in the view like showTypeErrorMessage().
As for _else_ branches, I think they are superfluous as you are calling cleanErrorMessages() at the validation start. This cleanup can be moved to onCreateClick() before presenter.createJob(...) is called.